hedera-node/src/main/java/com/hedera/services/txns/token/TokenCreateTransitionLogic.java

```diff
@@ -154,7 +156,17 @@ public ResponseCodeEnum validate(TransactionBody txnBody) {
 			return validity;
 		}
 
-		validity = initialSupplyAndDecimalsCheck(op.getInitialSupply(), op.getDecimals());
+		if (op.getTokenType() == TokenType.NonFungible) {
```

Comment:
I think that we should be checking against the representation type and not the token type.

Maybe we can add a new method for executing the `type` specific validation. That way the code won't have so many branches for the logic. 

Example: 
- Because the token is unique, it cannot have `initialSupply` != 0
- Because the token is non-fungible, it cannot have `decimals` != 0


We will have to add additional checks around `Common + NFT` or `Unique + Fungible` combinations and reject them with `NOT_SUPPORTED`